soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -122,7 +123,7 @@ void power_gate_entry(uint32_t core_id)
 	lpsheader->adsp_lpsram_magic = LPSRAM_MAGIC_VALUE;
 	lpsheader->lp_restore_vector = &dsp_restore_vector;
 	soc_cpus_active[core_id] = false;
-	xthal_dcache_all_writeback();
+	sys_cache_data_all(K_CACHE_WB);
```

Comment:
**_lpsheader_** is in the LPSRAM. Can you confirm that it will work? I have opened an issue on cache for this memory area https://github.com/zephyrproject-rtos/zephyr/issues/49112.

Reply:
@nashif Issue is fixed and xtensa hal function can be replace with _**z_xtensa_cache_flush_inv_all**_. 

I have applied your changes locally and unfortunately they break power gating functionality. Calling _**sys_cache_data_all(K_CACHE_WB)**_ leads to _**z_xtensa_cache_flush(NULL, Z_DCACHE_MAX)**_. It turns out that this function does not flush lpsram cache.

Reply:
Wasn't that the bug @edersondisouza fixes a few days ago?